opengever/maintenance/scripts/repository_migration.py

```diff
@@ -1179,16 +1246,27 @@ def main():
     PatchCommitSection()()
     PatchReindexContainersSection()()
     PatchReportSection()()
-    PatchTaskSyncWith()()
+    if options.sync_task:
+        PatchTaskSyncWith()()
+    else:
+        SkipTaskSyncWith()()
     PatchDisableLDAP()()
 
     logger.info('\n\nstarting analysis...\n')
-    analyser = RepositoryExcelAnalyser(mapping_path, options.output)
+    analyser = RepositoryExcelAnalyser(mapping_path, options.output_directory)
     analyser.analyse()
 
-    if options.output:
-        logger.info('\n\nwriting analysis excel...\n')
-        analyser.export_to_excel()
+    logger.info('\n\nwriting analysis excel...\n')
+    analyser.export_to_excel()
+
+    analyser_path = os.path.join(options.output_directory, "analyser.pkl")
+    with open(analyser_path, "w") as outfile:
+        # some attributes cannot be pickled so we simply set them to None
+        analyser.catalog = None
+        analyser.formatter = None
+        analyser.reporoot = None
+        analyser._reference_repository_mapping = None
+        pickle.dump(analyser, outfile)
```

Comment:
So, the only thing that you are interested here is the list of UIDs, right? I would advise against pickling the object. IMO you don't gain anything by the pickle, on the contrary it seems to be more complicated at first glance. Wouldn't something simpler like a text file with one item per line be sufficient here? The text file is human readable/editable and does not have any (ok, i admit, theoretical here) potential security impacts.

Reply:
Ok I now write some data that might be of interest to json files, as well of course as the list of task UIDs that we will need for syncing...